machine-learning-image-classification-with-javascript-and-nyckel/index.md

```diff
@@ -0,0 +1,338 @@
+    });
+
+    // file selected
+    $("#file").change(function ()
+    {
+        var file = $('#file')[0].files[0];
+        showImage(file);
+        resizeAndUploadImage(file);
+    });
+});
+// This function reads the uploaded file into your application.
+function showImage(file)
+{
+    var reader = new FileReader();
+    reader.onload = function (d)
+    {
+        $('.upload-area').css("background-image", "url(" + d.target.result + ")");
+    }
+    reader.readAsDataURL(file);
+}
+
+function resizeAndUploadImage(file)
+{
+    // Ensure it's an image (the file must be a photo)
+    if (file.type.match(/image.*/))
+    {
+        // Load the image
+        var reader = new FileReader();
+        reader.onload = function (readerEvent)
+        {
+            var image = new Image();
+            image.onload = function (imageEvent)
+            {
+                // this gives the specifications to the canvas about the measurements of the uploaded image
+                // It checks the image size and if it does not match the required dimensions, it is resized to fit the canvas
+                var canvas = document.createElement('canvas'),
+                    max_size = 600, //maximum size of any uploaded file
+                    width = image.width,
+                    height = image.height;
+                if (width > height)
+                {
+                    if (width > max_size)
+                    {
+                        height *= max_size / width;
+                        width = max_size;
+                    }
+                } else
+                {
+                    if (height > max_size)
+                    {
+                        width *= max_size / height;
+                        height = max_size;
+                    }
+                    //New dimensions are given to the image if it exceeds the defined maximum
+                }
+                canvas.width = width;
+                canvas.height = height;
+                canvas.getContext('2d').drawImage(image, 0, 0, width, height);
+                var dataUrl = canvas.toDataURL('image/jpeg');
+                var resizedImage = dataURLToBlob(dataUrl); //the resized image is then interpreted as raw data
+                
+                checkImageWithNyckel(resizedImage);
+            }
+            image.src = readerEvent.target.result;
+        }
+        //this ensures that the image is read by Nyckel as raw data
+        reader.readAsDataURL(file);
+    }
+    else
+    {
+    // If the file is not an image, upload is rejected with an alert and the page is reset to default
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+```
+The rectangular area for hosting the uploaded image is called a **canvas** in html. A **blob** on the other hand represents objects in form of raw data. Converting canvas to blob translates the images uploaded to your application in a way that the model can understand.

```

Comment:
Hi, could you show me the explanations for this code snippet? (Not comments!)

Reply:
@jumamiller this paragraph `The rectangular area for hosting the uploaded image is called a **canvas** in html. A **blob** on the other hand represents objects in form of raw data. Converting canvas to blob translates the images uploaded to your application in a way that the model can understand` gives the general objective of the code while the many comments I added gives the code explanation/objective where needed. Kindly check them out

Reply:
Please reconsider the comments